lib/chain.js

```diff
@@ -100,9 +100,7 @@ module.exports = class SetupChain {
       // remap array values with lookup result
       if (Array.isArray(value)) {
         out[key] = value.map((val) => {
-          if (typeOf(val) === 'object') return this.lookup(val)
-          const resolved = this[lookup](this.parse(val))
-          return resolved
+          return this.lookup(val)
         })
```

Comment:
What was the issue you were running into?

Reply:
Calling look on an object that mixes literal values and the string templates fails
```
lookup({
  account: "#account.account"
, position: {
    start: [0, 0]
  , end: [1, 1]
  }
})
```

The code as-is treats complex types as if they're strings, calling `this[lookup](this.parse(val))` which fails because it expects a string.

Reply:
I added a unit test which shows this failed case in the previous code

Reply:
ya, nice find. I can't believe no one has ever passed an array around that had something other than strings in it